src/tests/payment/test_stripe_webhook.py

```diff
@@ -0,0 +1,430 @@
+                "invoice_pdf": "https://stripe.com/invoice/test123.pdf",
+            }
+        },
+    }
+
+
+@pytest.fixture
+def customer_subscription_updated_payload():
+    return {
+        "type": "customer.subscription.updated",
+        "data": {
+            "object": {
+                "current_period_end": 1734440765,  # Timestamp for period end
+                "current_period_start": 1731848765,  # Timestamp for period start
+                "customer": "cus_REbNQXKKFCRF2c",
+                "cancel_at_period_end": False,
+                "trial_end": "null",  # No trial
+                "items": {
+                    "data": [
+                        {
+                            "plan": {
+                                "active": True,
+                            },
+                        },
+                    ],
+                },
+            }
+        },
+    }
+
+
+@pytest.fixture
+def checkout_session_completed_payload():
+    """Fixture to generate a 'checkout.session.completed' event payload."""
+    return {
+        "type": "checkout.session.completed",
+        "data": {
+            "object": {
+                "id": "cs_test_a1SdT2v81sae3cW",
+                "object": "checkout.session",
+                "amount_subtotal": 990,
+                "amount_total": 990,
+                "billing_address_collection": "required",
+                "cancel_url": "http://127.0.0.1:8000/api/v1/subscriptions/cancel/",
+                "created": 1731848739,
+                "currency": "eur",
+                "custom_fields": [],
+                "customer": "cus_REbNQXKKFCRF2c",
+                "customer_creation": "always",
+                "customer_details": {
+                    "address": {
+                        "city": "Saint-Ouen-l'Aumône",
+                        "country": "FR",
+                        "line1": "1 Rue Pagnere",
+                        "line2": "null",
+                        "postal_code": "95310",
+                        "state": "null",
+                    },
+                    "email": "kadi@test.com",
+                    "name": "kadi",
+                },
+                "customer_email": "kadi@test.com",
+                "expires_at": 1731935139,
+                "metadata": {"product_name": "Essentiel", "user_id": "5"},
+                "mode": "subscription",
+                "payment_status": "paid",
+                "status": "complete",
+            },
+        },
+    }
+
+
+def generate_stripe_signature(checkout_session_completed_payload, secret):
+    """
+    Simulates the Stripe signature for the test using the secret key.
+    """
+    payload_str = json.dumps(checkout_session_completed_payload)
+    timestamp = str(int(time.time()))
+    sig_base = f"{timestamp}.{payload_str}"
+    signature = hmac.new(
+        secret.encode("utf-8"), sig_base.encode("utf-8"), hashlib.sha256
+    ).hexdigest()
+    return f"t={timestamp},v1={signature}"
```

Comment:
I was wondering if we couldn't reuse some of `stripe._webhook.WebhookSignature`, but maybe not

Reply:
I don't think it's possible to reuse stripe._webhook.WebhookSignature, so I've implemented it manually.